spotlight-client/src/DataStore/UserStore.test.ts

```diff
@@ -181,3 +186,29 @@ test("passes target URL to callback", async () => {
   await store.authorize({ handleTargetUrl: callback });
   expect(callback.mock.calls[0][0]).toBe(targetUrl);
 });
+
+test("retrieves the state code from app_metadata and sets tenantStore's currentTenantId", async () => {
```

Comment:
It would be nice to have a test for the case when state_code is undefined.

Reply:
What is the state code for Recidiviz users? `recidiviz`? What happens in that case?

Reply:
I don't have a state code for recidiviz users right now, right now if you don't have a state_code set you see all states.

Reply:
Right now if your state code is `recidiviz` you would see no states. Should I handle it so that if your state code is set to `recidiviz` you see all states?

Reply:
If Recidiviz users get no state code then the undefined state code test should suffice.

What are all the state code options? In production the state_code is never set, staging state_code will be set for users from different states, or undefined for recidiviz users? You may want to document either with tests or comments somewhere what these different cases can be. I'm super confused reading this about what the possibilities are, and in which environment and what type of user takes which branch.

Reply:
Good point, I'm not sure where to document all the different ways the app gets locked to a certain state, maybe in the README?  And clearly comment all the tests?